diff -Nur vdr-1.3.27/vdr.c vdr-1.3.27-reel/vdr.c
--- vdr-1.3.27/vdr.c    2005-10-03 13:51:15.000000000 +0200
+++ vdr-1.3.27-reel/vdr.c       2005-09-10 18:09:22.000000000 +0200
@@ -121,6 +118,8 @@
   int WatchdogTimeout = DEFAULTWATCHDOG;
   const char *Terminal = NULL;
   const char *Shutdown = NULL;
+  bool TimerWakeup = false; 
+  bool AutoShutdown = false;
   cPluginManager PluginManager(DEFAULTPLUGINDIR);
   int ExitCode = 0;
 
@@ -140,6 +139,7 @@
       { "record",   required_argument, NULL, 'r' },
       { "shutdown", required_argument, NULL, 's' },
       { "terminal", required_argument, NULL, 't' },
+      { "timerwakeup", no_argument,    NULL, 'T' },
       { "version",  no_argument,       NULL, 'V' },
       { "video",    required_argument, NULL, 'v' },
       { "watchdog", required_argument, NULL, 'w' },
@@ -147,7 +147,7 @@
     };
 
   int c;
-  while ((c = getopt_long(argc, argv, "a:c:dD:E:hk:l:L:mp:P:r:s:t:v:Vw:", long_options, NULL)) != -1) {
+  while ((c = getopt_long(argc, argv, "a:c:dD:E:hk:l:L:mp:P:r:s:t:Tv:Vw:", long_options, NULL)) != -1) {
         switch (c) {
           case 'a': AudioCommand = optarg;
                     break;
@@ -224,6 +224,8 @@
                        return 2;
                        }
                     break;
+          case 'T': TimerWakeup = true;
+                    break;
           case 'V': DisplayVersion = true;
                     break;
           case 'v': VideoDirectory = optarg;
@@ -383,6 +385,7 @@
   int LastTimerChannel = -1;
   int PreviousChannel[2] = { 1, 1 };
   int PreviousChannelIndex = 0;
+  time_t vdrStartTime = time(NULL);
   time_t LastChannelChanged = time(NULL);
   time_t LastActivity = 0;
   int MaxLatencyTime = 0;
@@ -622,6 +625,14 @@
         if (time(NULL) - LastChannelChanged >= Setup.ZapTimeout && LastChannel != PreviousChannel[0] && LastChannel != PreviousChannel[1])
            PreviousChannel[PreviousChannelIndex ^= 1] = LastChannel;
         // Timers and Recordings:
+        if (TimerWakeup && Shutdown && time(NULL) - vdrStartTime > SHUTDOWNWAIT) { 
+           if (LastActivity == 0) {
+               LastActivity = 1;
+               AutoShutdown = true;
+           }
+           else if (LastActivity != 1)
+               AutoShutdown = false;
+           }
         if (!Timers.BeingEdited()) {
            // Delete expired timers:
            Timers.DeleteExpired();
@@ -891,7 +902,7 @@
                  Skins.Message(mtInfo, tr("Editing process finished"));
               }
            }
-        if (!Interact && ((!cRecordControls::Active() && !cCutter::Active() && (!Interface->HasSVDRPConnection() || UserShutdown)) || ForceShutdown)) {
+        if (!Interact && ((!cRecordControls::Active() && !cCutter::Active() && (!Interface->HasSVDRPConnection() || UserShutdown || AutoShutdown)) || ForceShutdown)) {
            time_t Now = time(NULL);
            if (Now - LastActivity > ACTIVITYTIMEOUT) {
               // Shutdown:
@@ -909,7 +920,7 @@
                     else
                        LastActivity = 1;
                     }
-                 if (UserShutdown && Next && Delta <= Setup.MinEventTimeout * 60 && !ForceShutdown) {
+                 if (UserShutdown && Next && Delta <= Setup.MinEventTimeout * 60 && !ForceShutdown && !AutoShutdown) {
                     char *buf;
                     asprintf(&buf, tr("Recording in %d minutes, shut down anyway?"), Delta / 60);
                     if (Interface->Confirm(buf))
@@ -918,8 +929,9 @@
                        UserShutdown = false;
                     free(buf);
                     }
-                 if (!Next || Delta > Setup.MinEventTimeout * 60 || ForceShutdown) {
+                 if (!Next || Delta > Setup.MinEventTimeout * 60 || ForceShutdown || (AutoShutdown && Delta > Setup.MinEventTimeout * 60)) {
                     ForceShutdown = false;
+                    AutoShutdown = false;
                     if (timer)
                        dsyslog("next timer event at %s", *TimeToString(Next));
                     if (WatchdogTimeout > 0)
@@ -973,6 +974,7 @@
                        isyslog("executing '%s'", cmd);
                        SystemExec(cmd);
                        free(cmd);
+                       Interrupted=1; // GA
                        }
                     else if (WatchdogTimeout > 0) {
                        alarm(WatchdogTimeout);
