diff -ur busybox-1.2.1-org/util-linux/mount.c busybox-1.2.1/util-linux/mount.c
--- busybox-1.2.1-org/util-linux/mount.c	2006-07-01 00:42:13.000000000 +0200
+++ busybox-1.2.1/util-linux/mount.c	2006-10-24 22:21:17.000000000 +0200
@@ -291,7 +291,7 @@
 	// Look at the file.  (Not found isn't a failure for remount, or for
 	// a synthetic filesystem like proc or sysfs.)
 
-	if (lstat(mp->mnt_fsname, &st));
+	if (stat(mp->mnt_fsname, &st));
 	else if (!(vfsflags & (MS_REMOUNT | MS_BIND | MS_MOVE))) {
 		// Do we need to allocate a loopback device for it?
 
