--- keybdev.c.org	Thu Jun 26 00:10:52 2003
+++ keybdev.c	Mon Oct 13 13:58:45 2003
@@ -191,6 +191,9 @@ static struct input_handle *keybdev_conn
 	if (i == BTN_MISC)
  		return NULL;
 
+	if (dev->name[0]=='D' && dev->name[1]=='V' && dev->name[2] == 'B')
+		return NULL;
+
 	if (!(handle = kmalloc(sizeof(struct input_handle), GFP_KERNEL)))
 		return NULL;
 	memset(handle, 0, sizeof(struct input_handle));
