diff -Nru vdr-1.3.21-vanilla/menuitems.h vdr-1.3.21-enaio/menuitems.h
--- vdr-1.3.21-vanilla/menuitems.h	2004-11-21 15:23:00.000000000 +0200
+++ vdr-1.3.21-enaio/menuitems.h	2005-02-13 18:22:50.017988456 +0200
@@ -82,6 +82,9 @@
   char *allowed;
   int pos;
   bool insert, newchar, uppercase;
+  int ieCurChr;
+  eKeys lastKey;
+  time_t lastActivity;
   void SetHelpKeys(void);
   virtual void Set(void);
   char Inc(char c, bool Up);
